cpp/include/raft/stats/r2_score.cuh

```diff
@@ -45,6 +46,33 @@ math_t r2_score(math_t* y, math_t* y_hat, int n, cudaStream_t stream)
   return detail::r2_score(y, y_hat, n, stream);
 }
 
+/**
+ * Calculates the "Coefficient of Determination" (R-Squared) score
+ * normalizing the sum of squared errors by the total sum of squares.
+ *
+ * This score indicates the proportionate amount of variation in an
+ * expected response variable is explained by the independent variables
+ * in a linear regression model. The larger the R-squared value, the
+ * more variability is explained by the linear regression model.
+ *
+ * @tparam value_t the data type
+ * @tparam idx_t index type
+ * @param handle the raft handle
```

Comment:
Please add `[in]`, `[inout]` and `[out]` to the docs for all params (applies for all changes in this PR).